### DISCUSSION TITLE
ESP32 on Tasmota BLE 10.1.0 + 3 relays running unstable ... question regarding power

### DISCUSSION THREAD

**Discussion by jenshenk** [arendst/Tasmota#14793] [Q&A] [answered]
Hi,
I've an issue that is really hard to track down since it is very inconsistent

To an ESP32 running Tasmota BLE 10.1.0 I have 3 relays directly connected via the ESP's PINs. The relays are 5V ones and the ESP32 is powered via a 2A source on USB.  
 
Recently I had an power outage, afterwards I needed to unplug/plug the esp once a day for 3 days. Guess it hung in a reboot loop, the red led kept blinking but it never got online. Afterwards the same setup ran continuously, now has an online time of 6 days.

Eliminating all reasons I could think of the last thing that came to mind (should have been the first one I guess :)) was the power situation. I now keep wondering whether it would be better to power the relays directly from the source and not via the ESP PINs. On the other hand I now see this setup running since days and doing its job so I would love to get some input whether changing around the wiring is actually necessary since how I put it together it would mean quite some work. 

Thanks in advance!
Jens

**Comment by Jason2866** [comment]
What do you mean power the relais via GPIOs? There has to be always a driver circuit (transistor)
to drive a relais.

**Reply by jenshenk** [reply]
Thanks for the quick reaction... Guess I wasn't correct with my wording: The relays are on little PCBs that include this, so they are ready to use modules.

**Comment by Jason2866** [comment, marked as answer]
Powering the relais directly from the 5V power supply is always a better idea than routing over the esp circuit board.

**Reply by jenshenk** [reply]
It already felt like this is the case :) Just curious though: Is it possible that this random hang at reboot is due to the whole power situation being right at the edge and therefore often enough yet just sometimes not?

**Comment by jenshenk** [comment]
For people stumbling upon this: Tasmota 11.0 seems to be much more forgiving for tricky power situations, after upgrading all devices run without restarts now. Seems in connection with power/restarts there was in general a tricky period between 9.5 and 11.0 based on a couple of discussions I found.
